src/test/kotlin/it/unibo/protelis/web/execution/ProtelisUpdateMessageTest.kt

```diff
@@ -0,0 +1,69 @@
+package it.unibo.protelis.web.execution
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.vertx.core.json.Json
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+
+class ProtelisUpdateMessageTest {
+
+  @Test
+  @DisplayName("Test ProtelisUpdateMessage comparisons")
+  fun `test ProtelisUpdateMessage`() {
+    val id1 = "Pippo"
+    val id2 = "Pluto"
+    val coordinates1 = 0 to 1
+    val coordinates2 = 1 to 0
+    val n1 = ProtelisNode(id1, coordinates1)
+    val n2 = ProtelisNode(id2, coordinates2)
+    val ue = ProtelisUpdateMessage(listOf())
+    val u1 = ProtelisUpdateMessage(listOf(n1))
+    val u2 = ProtelisUpdateMessage(listOf(n1, n2))
+    assertEquals(ProtelisUpdateMessage(listOf(n1, n2)), u2)
+    assertEquals(ProtelisUpdateMessage(listOf()), ue)
+    assertNotEquals(ProtelisUpdateMessage(listOf(n2)), u1)
+    assertNotEquals(ue, u1)
+    assertEquals(ProtelisUpdateMessage(listOf(n1)).hashCode(), u1.hashCode())
+  }
+
+  @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [UndocumentedPublicFunction - [test JSON serialization] at /src/src/test/kotlin/it/unibo/protelis/web/execution/ProtelisUpdateMessageTest.kt:31:3](https://app.codacy.com/manual/NiccoMlt/Protelis-Web/pullRequest?prid=4969516)